spec/listformat.html

```diff
@@ -183,15 +183,19 @@ <h1>Intl.ListFormat ( [ _locales_ [ , _options_ ] ] )</h1>
         1. Else
           1. Let _options_ be ? ToObject(_options_).
         1. Let _opt_ be a new Record.
-        1. Let _type_ be GetOption(_options_, `"type"`, `"string"`, &laquo; `"conjunction"`, `"disjunction"`, `"unit"` &raquo;, `"conjunction"`).
-        1. Set _listFormat_.[[Type]] to _type_.
-        1. Let _style_ be GetOption(_options_, `"style"`, `"string"`, &laquo; `"long"`, `"short"`, `"narrow"` &raquo;, `"long"`).
-        1. Set _listFormat_.[[Style]] to _style_.
-        1. Let _localeData_ be %ListFormat%.[[LocaleData]].
         1. Let _matcher_ be ? GetOption(_options_, `"localeMatcher"`, `"string"`, &laquo; `"lookup"`, `"best fit"` &raquo;, `"best fit"`).
         1. Set _opt_.[[localeMatcher]] to _matcher_.
-        1. If _style_ is `"narrow"` and _type_ is not `"unit"`, throw a *RangeError* exception.
+        1. Let _localeData_ be %ListFormat%.[[LocaleData]].
         1. Let _r_ be ResolveLocale(%ListFormat%.[[AvailableLocales]], _requestedLocales_, _opt_, %ListFormat%.[[RelevantExtensionKeys]], _localeData_).
+        1. Set _listFormat_.[[Locale]] to _r_.[[locale]].
+        1. Let _type_ be ? GetOption(_options_, `"type"`, `"string"`, &laquo; `"conjunction"`, `"disjunction"`, `"unit"` &raquo;, `"conjunction"`).
+        1. Set _listFormat_.[[Type]] to _type_.
+        1. If _type_ is `"unit"`,
+          1. Let _allowedStyles_ be &laquo; `"long"`, `"short"`, `"narrow"` &raquo;.
+        1. Else,
+          1. Let _allowedStyles_ be &laquo; `"long"`, `"short"` &raquo;.
```

Comment:
Did we discuss that? It looks like we're leaking the internal CLDR data structure here. Sometimes I see a reason to do that (CLDR is well thought through), but sometimes the lack of a particular field is just a temporary omission and other data sources could provide `narrow` for other `type`.

Reply:
This is just a different way to spec than the having the following line after getting the style. Both will throw the RangeError under the same condition. 
> If _style_ is `"narrow"` and _type_ is not `"unit"`, throw a *RangeError* exception.